src/cart/crt_hg.h

```diff
@@ -193,11 +194,11 @@ crt_hgret_2_der(int hg_ret)
 	case HG_BUSY:
 		return -DER_BUSY;
 	case HG_FAULT:
-	case HG_PROTOCOL_ERROR:
 		return -DER_HG_FATAL;
 	case HG_PERMISSION:
 	case HG_ACCESS:
 		return -DER_NO_PERM;
+	case HG_PROTOCOL_ERROR:
```

Comment:
While I'm okay with this, I think a better solution to assess in what cases Mercury should not return this error. See [my comment today on the Jira ticket](https://daosio.atlassian.net/browse/DAOS-15044?focusedCommentId=124785).

Reply:
Or, perhaps these two approaches are both necessary...

Reply:
As per discussion with Jerome, this error is often returned by mercury as a result of 'unknown error' type in the lower levels (fabric and ucx), such as something failed but didnt specify the reason. As such this error might be retriable in some contexts (such as send failure) but not in others (such as bulk allocation failure).